ecs-cli/modules/cli/local/project/project.go

```diff
@@ -0,0 +1,214 @@
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/cli/local/converter"
+	ecsclient "github.com/aws/amazon-ecs-cli/ecs-cli/modules/clients/aws/ecs"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/clients/aws/secretsmanager"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/commands/flags"
+	"github.com/aws/amazon-ecs-cli/ecs-cli/modules/config"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/urfave/cli"
+)
+
+const (
+	LocalOutDefaultFileName = "./docker-compose.local.yml"
+	LocalOutFileMode        = os.FileMode(0644) // Owner=read/write, Other=readonly
+	LocalInFileName         = "./task-definition.json"
+)
+
+type LocalProject interface {
+	ReadTaskDefinition() error
+	Convert() error
+	Write() error
+	LocalOutFileName() string
+}
+
+type localProject struct {
+	context          *cli.Context
+	taskDefinition   *ecs.TaskDefinition
+	localBytes       []byte
+	localOutFileName string
+}
+
+func New(context *cli.Context) LocalProject {
+	p := &localProject{context: context}
+	return p
+}
+
+func (p *localProject) TaskDefinition() *ecs.TaskDefinition {
+	return p.taskDefinition
+}
+
+func (p *localProject) LocalOutFileName() string {
+	return p.localOutFileName
+}
+
+func (p *localProject) ReadTaskDefinition() error {
+	arn := p.context.String(flags.TaskDefinitionArnFlag)
+	filename := p.context.String(flags.TaskDefinitionFileFlag)
+
+	if arn != "" && filename != "" {
+		return fmt.Errorf("Cannot specify both --%s and --%s flags.", flags.TaskDefinitionArnFlag, flags.TaskDefinitionFileFlag)
+	}
+
+	var taskDefinition *ecs.TaskDefinition
+	var err error
+
+	if arn != "" {
+		taskDefinition, err = p.readTaskDefinitionFromArn(arn)
+		if err != nil {
+			return err
+		}
+	} else if filename != "" {
+		taskDefinition, err = p.readTaskDefinitionFromFile(filename)
+		if err != nil {
+			return err
+		}
+	} else if _, err := os.Stat(LocalInFileName); err == nil {
+	// Try reading local task-definition.json file by default
+		taskDefinition, err = p.readTaskDefinitionFromFile(LocalInFileName)
+		if err != nil {
+			return err
+		}
+	}
+
+	if taskDefinition == nil {
+		return fmt.Errorf("Could not detect valid Task Definition")
+	}
+
+	p.taskDefinition = taskDefinition
+	return nil
+}
+
+func (p *localProject) readTaskDefinitionFromFile(filename string) (*ecs.TaskDefinition, error) {
+	bytes, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, fmt.Errorf("Error reading task definition from %s: %s", filename, err.Error())
+	}
+
+	taskDefinition := ecs.TaskDefinition{}
+	err = json.Unmarshal(bytes, &taskDefinition)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing task definition JSON: %s", err.Error())
+	}
+
+	return &taskDefinition, nil
+}
+
+var newCommandConfig = func(context *cli.Context, rdwr config.ReadWriter) (*config.CommandConfig, error) {
+	return config.NewCommandConfig(context, rdwr)
+}
+
+// FIXME: NOTE this will actually read from either ARN or Task Definition family name
```

Comment:
Is this FIXME to just rename the function and parameter or is there something else that's more involved?

If it's naming maybe `readTaskDefinitionFromRemote(remote string)` with a comment saying "remote is the ARN or family".  Or, `readTaskDefinitionFromARNOrFamily(arnOrFamily string)` it's a mouthful but maybe more explicit (ugh.)

Reply:
I think I was thinking the renaming would apply to the flag itself, and referring to the parameter as "Id"?

Reply:
👍 sgtm

Reply:
Similarly, in @iamhopaul123 he added a new flag: https://github.com/aws/amazon-ecs-cli/pull/789/files#diff-9c07acb4867a9b5d46a004942a378e76R153